### PR TITLE
Feat: Initial Set Up, All Trips Endpoint, Trip Show Endpoint

### PR DESCRIPTION
**Pull request by IgorrMagalhaess**
## Type of change
- [x] New feature
- [ ] Bug Fix
- [x] Test
- [ ] Refactor/Style

### Initial set up:
Gems (Faker, Factory-Bot, SimpleCov, rspec-rails, capybara, launch, webmock, VCR, shoulda-matchers)
Models (Trip, Accomodation, Activity, Daily_Itinerary)
Routes
DB tables

Relevant User Story(s):
GET '/api/v1/trips' - Get all trips for specific user
GET '/api/v1/trips/:id' - Get a specific trip for specific user

*Brief summary of the change(s) included in this pull request*
This PR includes the initial setup, including database, migrations, gems and two endpoints for our backend API. 

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [x] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*
**All Tests Passing**

## Checklist:
- [x] My code has no unused/commented out code
- [ ] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

## Fun fact or favorite .gif:
GIF

### CODE REVIEW

File: db/migrate/20240409220920_create_accomodations.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateAccomodations < ActiveRecord::Migration[7.1]
+  def change
+    create_table :accomodations do |t|
+      t.references :trip, null: false, foreign_key: true
+      t.string :address
+      t.float :lat
+      t.float :lon
+      t.integer :type_of_accomodation
+
```
Comment (yainporter) [inline db/migrate/20240409220920_create_accomodations.rb:9]: Did we also want to add check in and check out as well here?

File: app/serializers/trips_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class TripsSerializer
+  include JSONAPI::Serializer
```
Comment (yainporter) [inline app/serializers/trips_serializer.rb:2]: I think this should be TripSerializer, not Trip(s)?

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -62,4 +65,21 @@
...
+  end
 end
+
+VCR.configure do |config|
+  config.cassette_library_dir = "spec/fixtures/vcr_cassettes"
+  config.hook_into :webmock
+  # config.filter_sensitive_data('<Authorization>') { Rails.application.credentials.tomtom[:key]}
+  # config.default_cassette_options = { re_record_interval: 7.days }
+  # config.configure_rspec_metadata!
+end
```
Comment (yainporter) [inline spec/rails_helper.rb:85]: I'd like to make a note to remember to remove this if we don't end up consuming an API on our back end

File: Gemfile [hunk 4]
```diff
@@ -31,6 +31,9 @@ gem "bootsnap", require: false
 
 # Use Active Storage variants [https://guides.rubyonrails.org/active_storage_overview.html#transforming-images]
 # gem "image_processing", "~> 1.2"
+gem "faraday"
+
```
Comment (yainporter) [inline Gemfile:35]: Just leaving this for a note to remove if we don't end up consuming an API

**Review by yainporter — CHANGES_REQUESTED** [review]
Looks great so far, Igor! Let's make a note to do some error handling on these.

Also, if you use this command: `git rm --cached -r coverage/` it will delete all the simplecov files

File: db/migrate/20240409215614_create_activities.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class CreateActivities < ActiveRecord::Migration[7.1]
+  def change
+    create_table :activities do |t|
+      t.string :address
+      t.string :description
+      t.float :lat
+      t.float :lon
+      t.string :type
```
Comment (yainporter) [inline db/migrate/20240409215614_create_activities.rb:8]: I am making a seed file to test the API in Postman, it looks like ActiveRecord does not like us naming a variable `type`, so we will need to change this to maybe `activity_type` instead

**Review by yainporter — CHANGES_REQUESTED** [review]

File: app/serializers/trips_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+  attributes :name, :location, :start_date, :end_date, :status, :total_budget
+end
```
Comment (yainporter) [inline app/serializers/trips_serializer.rb:4]: Do we also want to add the method that will calculate the total_expense here?

Comment (hoadam) [reply app/serializers/trips_serializer.rb:4]: Agree with Yain, we will need an attribute total_expenses

File: db/migrate/20240409220920_create_accomodations.rb (hunk 1 above)
Comment (hoadam) [reply db/migrate/20240409220920_create_accomodations.rb:9]: Agree with Yain, I think we will need a checkin checkout

**Comment by hoadam** [comment]
Thank you for your work Igor! And I would suggest you add coverage into .gitignore, so you don't have to delete it everytime.

File: db/migrate/20240409220920_create_accomodations.rb (hunk 1 above)
Comment (IgorrMagalhaess) [reply db/migrate/20240409220920_create_accomodations.rb:9]: Sounds good to me!!

File: app/serializers/trips_serializer.rb (hunk 2 above)
Comment (IgorrMagalhaess) [reply app/serializers/trips_serializer.rb:2]: Nice catch!! I'll change it!

**Pull request closed without merging**